File: wheels/2.实用功能封装/2.3objectKeyValues/objectKeyValues/ViewController.m
//
//  ViewController.m
//  objectKeyValues
//
//  Created by WZH on 16/8/12.
//  Copyright © 2016年 王振海. All rights reserved.
//

#import "ViewController.h"
#import "NSObject+keyValues.h"
#import "student.h"
#import "School.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    NSMutableArray *sts = [NSMutableArray array];
//    student *st = [[student alloc] init];
//    st.name = @"wzh";
//    [sts addObject:st];
//    student *st2 = [[student alloc] init];
//    st2.name = @"wss";
//    [sts addObject:st2];
    
    School *school = [[School alloc] init];
//    school.age = @(100);
//    school.name = @"要塞";
//    school.st = sts;
    school.bo = YES;
    school.doub = 80.1000;
    school.inter = 10;
    school.Uinter = 20;
    school.cgfloat = 1000;
    school.f  = 2000.00;
    
    NSLog(@"%@",school.keyValues);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Demo/Swift/语法/26.Swift3.0 - 懒加载/26.Swift3.0 - 懒加载/ViewController.swift
//
//  ViewController.swift
//  26.Swift3.0 - 懒加载
//
//  Created by WZH on 16/10/19.
//  Copyright © 2016年 limpid. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    //只是定义，没有分配空间和初始化
    
    //什么时候用as转换 -> 父类转换为子类
    //初始化并且分配控件，会提前创建
    //移动端开发，“延迟加载”减少内存消耗
    //懒加载 - lazy
    //最大的好处 ->解除解包烦恼
//    lazy var label: DemoLabel =  DemoLabel()
    
    //懒加载本质上是一个闭包
    //完整写法： 供参考 不建议这么写
    //{}包装代码  （）执行代码
    lazy var label = { () -> DemoLabel in
        
        let l = DemoLabel()
        
        //设置label的属性
        
        return l
        
    }()
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setUp()
    }
    
    private func setUp() {
        
        
        //2.添加到视图
        //! 解包，为了参与计算， addSubView，用subviews 数组记录空间，数组中不允许加入nil
        //? ---- 可选解包，调用方法，如果为nil，不调用方法，但是不能参与计算
        view.addSubview(label)
        
        label.text = "hello"
        label.sizeToFit()
        label.center = view.center
    }


}

